demos/interactive_face_detection_demo/main.cpp

```diff
@@ -59,340 +69,519 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+                                  {"sad", em_data[2]} ,
+                                  {"surprise", em_data[3]},
+                                  {"anger", em_data[4]}
+                                  });
         }
 
-        Visualizer::Ptr visualizer;
-        if (!FLAGS_no_show || !FLAGS_o.empty()) {
-            visualizer = std::make_shared<Visualizer>(cv::Size(width, height));
-            if (!FLAGS_no_show_emotion_bar && emotionsDetector.enabled()) {
-                visualizer->enableEmotionBar(emotionsDetector.emotionsVec);
+        if (landmarks_enable) {
+            const float *lm_data = out_landmarks[i].ptr<float>();
+
+            face->landmarksEnable();
+            std::vector<float> normedLandmarks;
+            size_t n_lm = 70;
+            for (size_t i_lm = 0UL; i_lm < n_lm; ++i_lm) {
+                normedLandmarks.push_back(lm_data[2 * i_lm]);
+                normedLandmarks.push_back(lm_data[2 * i_lm + 1]);
             }
+
+            face->updateLandmarks(normedLandmarks);
         }
+        // End of face postprocesing
 
-        // Detecting all faces on the first frame and reading the next one
-        faceDetector.enqueue(frame);
-        faceDetector.submitRequest();
+        faces.push_back(face);
+    }
+}
 
-        prev_frame = frame.clone();
+int main(int argc, char *argv[]) {
+    try {
+        std::cout << "InferenceEngine: " << GetInferenceEngineVersion() << std::endl;
 
-        // Reading the next frame
-        frameReadStatus = cap.read(frame);
+        // ------------------------------ Parsing and validating of input arguments --------------------------
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+        slog::info << "Start inference " << slog::endl;
+
+        std::ostringstream out;
+        size_t framesCounter = 0;
+        cv::Mat prev_frame, next_frame;
+        std::list<Face::Ptr> faces;
+        size_t id = 0;
 
         std::cout << "To close the application, press 'CTRL+C' here";
         if (!FLAGS_no_show) {
             std::cout << " or switch to the output window and press any key";
         }
         std::cout << std::endl;
 
-        while (true) {
-            timer.start("total");
-            framesCounter++;
-            isLastFrame = !frameReadStatus;
-
-            // Retrieving face detection results for the previous frame
-            faceDetector.wait();
-            faceDetector.fetchResults();
-            auto prev_detection_results = faceDetector.results;
-
-            // No valid frame to infer if previous frame is the last
-            if (!isLastFrame) {
-                faceDetector.enqueue(frame);
-                faceDetector.submitRequest();
-            }
+        bool age_gender_enable = !FLAGS_m_ag.empty();
+        bool headpose_enable   = !FLAGS_m_hp.empty();
+        bool emotions_enable   = !FLAGS_m_em.empty();
+        bool landmarks_enable  = !FLAGS_m_lm.empty();
```

Comment:
I'd also check here flags which exist but we don't support, and write a WARNING on these options.

Reply:
Done

Reply:
Ok